flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/batch/MatchRecognizeBatchRestoreTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.batch;
+
+import org.apache.flink.table.planner.plan.nodes.exec.common.MatchRecognizeTestPrograms;
+import org.apache.flink.table.planner.plan.nodes.exec.testutils.BatchRestoreTestBase;
+import org.apache.flink.table.test.program.TableTestProgram;
+
+import java.util.Arrays;
+import java.util.List;
+
+/** Batch Compiled Plan tests for {@link BatchExecMatch}. */
+public class MatchRecognizeBatchRestoreTest extends BatchRestoreTestBase {
+
+    public MatchRecognizeBatchRestoreTest() {
+        super(BatchExecMatch.class);
+    }
+
+    @Override
+    public List<TableTestProgram> programs() {
+        return Arrays.asList(
+                MatchRecognizeTestPrograms.MATCH_SIMPLE,
```

Comment:
Why are tests like `MATCH_ORDER_BY_EVENT_TIME` not part of this list? All tests currently only test processing-time but no test tests event-time.

Reply:
The ordering for the existing event time tests use data that has out-of-order elements: https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/MatchRecognizeTestPrograms.java#L132-L133

That means that the result sets are going to be different in batch mode.  

I see two options:
1. Provide separate results in TableTestPrograms when batch would return different results.
2. Add additional test programs.

I'd prefer 1 since this would also handle the situation where changelogs need to be materialized.

Reply:
From a chat with Timo, we decided that `common` TableTestPrograms should not contain late data so that streaming and batch execution modes return the same thing.

Reply:
Plan:  Rename the existing streaming test to have "with-late-data" or similar in the name.

Create new event time test cases without late data for use with Match Recognize.